server/utils.go

```diff
@@ -45,7 +45,7 @@ func PostJSONWithTimeout(ctx context.Context, url string, v interface{}) error {
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
+	if !(resp.StatusCode >= 200 && resp.StatusCode <= 299) {
```

Comment:
I have a feeling we may fiddle with this sort of thing in the future. Can we extract this into a helper `func httpSuccessStatus(int status) bool` or something similar?